bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusBinding.java

```diff
@@ -113,23 +112,24 @@ protected void internalUpdateItem(String slaveName, InputRegister[] registers, S
                 continue;
             }
 
-            String slaveValueType = modbusSlaves.get(slaveName).getValueType();
-            double rawDataMultiplier = modbusSlaves.get(slaveName).getRawDataMultiplier();
+            ModbusSlave slave = modbusSlaves.get(slaveName);
+            String slaveValueType = slave.getValueType();
+            double rawDataMultiplier = slave.getRawDataMultiplier();
 
             State newState = extractStateFromRegisters(registers, config.readIndex, slaveValueType);
             /* receive data manipulation */
             State newStateBoolean = provider.getConfig(itemName)
                     .translateBoolean2State(!newState.equals(DecimalType.ZERO));
             if (!UnDefType.UNDEF.equals(newStateBoolean)) {
                 newState = newStateBoolean;
-            } else if ((rawDataMultiplier != 1) && (config.getItem() instanceof NumberItem)) {
+            } else if ((rawDataMultiplier != 1) && (config.getState() instanceof DecimalType)) {
```

Comment:
Great!  Thanks @Wolfgang1966.  If your own testing confirms all good, then I think this PR is good to merge.